### DISCUSSION TITLE
Why are floating versions prohibited when using central package version management?

### DISCUSSION THREAD

**Discussion by voroninp** [NuGet/Home#12200] [Q&A]
I was surprised NuGet claims floating versions cannot be used for centrally managed packages. 

What's the rationale behind this?

**Comment by csi-lund** [comment]
I also was surprised by this - I wanted to do something like 

[1.*,2) to automatically take the latest non breaking change but also got the warning that floating versions were not supported.

**Comment by TheXenocide** [comment]
I get the impression that someone has decided that `.*` versions aren't useful or something. There seems to be very little support for them as new features are developed, yet we find them to be hugely helpful in our development processes. The `Pack` target and nuspec format seem uninterested in supporting "take the latest" styles of behavior, which tends to be somewhat inconvenient for us and results in a fair amount of rebuilding just to update nuspecs and explicit transitive version overriding that would be a lot easier if we had an alternative to taking a fixed or lowest matching version.

**Comment by TheXenocide** [comment]
Sidenote: I suspect one of the challenge areas here is around supporting `CentralPackageTransitivePinningEnabled` due to the dynamic nature of the floating version, which I think is understandable. That said, it seems like there could be a compromise that allows `.*` to be used for some packages and transitive pinning to be used for other packages that don't use `.*` (this would be better than an either or scenario, for us anyway)

**Comment by nkolev92** [comment]
https://github.com/NuGet/Home/issues/10432 and https://learn.microsoft.com/en-us/nuget/reference/errors-and-warnings/nu1011 answer these questions.
